From a83a936cd81d916b9caaa0415bd3330f44937921 Mon Sep 17 00:00:00 2001
From: Afzal Mohammed <afzal@ti.com>
Date: Sat, 30 Jun 2012 22:45:06 +0800
Subject: [PATCH 534/609] cpufreq: OMAP: am33xx support

Original commit: 8e4183b04b34e042cfa967cb56cdbed9bf9470df

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

am33xx mpu pll clock name is "dpll_mpu_ck",
same as that of omap44xx()

Signed-off-by: Afzal Mohammed <afzal@ti.com>
---
 drivers/cpufreq/omap-cpufreq.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/cpufreq/omap-cpufreq.c b/drivers/cpufreq/omap-cpufreq.c
index 6d04f31..7576506 100644
--- a/drivers/cpufreq/omap-cpufreq.c
+++ b/drivers/cpufreq/omap-cpufreq.c
@@ -314,9 +314,9 @@ static int __init omap_cpufreq_init(void)
 {
 	if (cpu_is_omap24xx())
 		mpu_clk_name = "virt_prcm_set";
-	else if (cpu_is_omap34xx())
+	else if (cpu_is_omap34xx() && !cpu_is_am33xx())
 		mpu_clk_name = "dpll1_ck";
-	else if (cpu_is_omap44xx())
+	else if (cpu_is_omap44xx() || cpu_is_am33xx())
 		mpu_clk_name = "dpll_mpu_ck";
 
 	if (!mpu_clk_name) {
-- 
1.7.5.4

